firstpartydata/first_party_data_test.go

```diff
@@ -0,0 +1,552 @@
+  				  "id":"reqSiteId",
+				  "content": {
+					"data":[
+						{ 
+						  "id": "siteDataId1",
+						  "name": "siteDataName1"
+						},
+						{
+ 						  "id": "siteDataId2",
+            			  "name": "siteDataName2"
+						}
+					]
+				  }
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M",
+				  "data":[
+						{ 
+						  "id": "userDataId1",
+						  "name": "userDataName1"
+						}
+					]
+  				},
+  				"app": {
+  				  "id": "appId",
+					"content":{
+						"data": [
+							{ 
+							  "id": "appDataId1",
+							  "name": "appDataName1"
+							}
+						]
+					}
+  				}
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id":"reqSiteId",
+				  "content": {}
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M"
+  				},
+  				"app": {
+  				  "id": "appId",
+				  "content": {}
+  				}
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: {{ID: "siteDataId1", Name: "siteDataName1"}, {ID: "siteDataId2", Name: "siteDataName2"}},
+				userDataKey:        {{ID: "userDataId1", Name: "userDataName1"}},
+				appContentDataKey:  {{ID: "appDataId1", Name: "appDataName1"}},
+			},
+		}, {
+			description: "User only data present",
+			input: []byte(`{
+  				"id": "bid_id",
+			 	"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id":"reqSiteId"
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M",
+				  "data":[
+						{ 
+						  "id": "userDataId1",
+						  "name": "userDataName1"
+						}
+					]
+  				},
+  				"app": {
+  				  "id": "appId"
+  				}
+			}`),
+			output: []byte(`{
+  				"id": "bid_id",
+				"imp":[{"id":"impid"}],
+  				"site": {
+  				  "id":"reqSiteId"
+  				},
+  				"user": {
+  				  "id": "reqUserID",
+  				  "yob": 1982,
+  				  "gender": "M"
+  				},
+  				"app": {
+  				  "id": "appId"
+  				}
+			}`),
+			expectedFpdData: map[string][]openrtb2.Data{
+				siteContentDataKey: nil,
+				userDataKey:        {{ID: "userDataId1", Name: "userDataName1"}},
+				appContentDataKey:  nil,
+			},
+		},
+	}
+	for _, test := range testCases {
+
+		var req openrtb2.BidRequest
+		err := json.Unmarshal(test.input, &req)
+		assert.NoError(t, err, "Error should be nil")
+
+		res := ExtractOpenRtbGlobalFPD(&req)
+
+		resReq, err := json.Marshal(req)
+		assert.NoError(t, err, "Error should be nil")
+
+		assert.JSONEq(t, string(test.output), string(resReq), "Result request is incorrect")
+		assert.Equal(t, test.expectedFpdData[siteContentDataKey], res[siteContentDataKey], "siteContentData data is incorrect")
+		assert.Equal(t, test.expectedFpdData[userDataKey], res[userDataKey], "userData is incorrect")
+		assert.Equal(t, test.expectedFpdData[appContentDataKey], res[appContentDataKey], "appContentData is incorrect")
+
+	}
+}
+
+func TestPreprocessFPD(t *testing.T) {
```

Comment:
Please rename this test to be: `TestExtractBidderConfigFPD` as it is testing `ExtractBidderConfigFPD` function

Reply:
Renamed